### PR TITLE
Add support for slash commands and interactive msg

### PR DESCRIPTION
**Pull request by k321rf**
Add support for slash commands within CommandDefinition.

Add support for interactive events as part of the CommandDefinition.

```golang

func wasArticleUseFulCmd(actionId string) func(botCtx slacker.BotContext, request slacker.Request, response slacker.ResponseWriter) {
	return func(botCtx slacker.BotContext, request slacker.Request, response slacker.ResponseWriter) {
        ....
        }
}

func wasArticleUseFulInteractive(s *slacker.Slacker, e *socketmode.Event, callback *slack.InteractionCallback, request *socketmode.Request) {
...
}

func main() {
  bot := slacker.NewClient(....)
  bot.Command("was-this-article-useful", &slacker.CommandDefinition{
		  BlockID:     "was_article_useful",
		  Handler:     wasArticleUseFulCmd("was_article_useful"),
		  Interactive: wasArticleUseFulInteractive,
	  })
}
```

### CODE REVIEW

**Comment by k321rf** [comment]
This PR is similar to https://github.com/shomali11/slacker/pull/88.

**Comment by ctennis** [comment]
I'm happy to let this one go forward and redact my proposed PR :)

**Comment by arusso** [comment]
I’ve been meaning to review #88 again but have been pulled away by life for
a bit. I’ll check both out this week and see about getting slash command
support merged in.

**Comment by mrmm** [comment]
Hello @arusso, I hope that everything went great with the review. Is there a way to help move this PR forward?

File: context.go [hunk 1]
```diff
@@ -55,9 +55,15 @@ type MessageEvent struct {
 	// Channel ID where the message was sent
 	Channel string
 
+	// ChannelName where the message was sent
+	ChannelName func() string
```
Comment (arusso) [inline context.go:59]: All the other struct fields are strings, for the sake of consistency ChannelName and UserName should be as well.

File: context.go [hunk 2]
```diff
@@ -55,9 +55,15 @@ type MessageEvent struct {
...
+
 	// User ID of the sender
 	User string
 
+	// UserName of the the sender
+	UserName func() string
```
Comment (arusso) [inline context.go:65]: This should be a a `string` field.

**Review by arusso — CHANGES_REQUESTED** [review]
Thanks for the submission!

Could you provide a minimal example that one can run to exercise the code to go with this?

Once you make the field changes and provide a minimal example we can finally get this merged.

Thanks!

File: slacker.go [hunk 3]
```diff
@@ -363,14 +380,22 @@ func (s *Slacker) handleMessageEvent(ctx context.Context, evt interface{}) {
...
+func newMessageEvent(slacker *Slacker, evt interface{}, req *socketmode.Request) *MessageEvent {
 	var me *MessageEvent
 
 	switch ev := evt.(type) {
 	case *slackevents.MessageEvent:
 		me = &MessageEvent{
-			Channel:         ev.Channel,
-			User:            ev.User,
+			Channel: ev.Channel,
+			ChannelName: func() string {
```
Comment (k321rf) [inline slacker.go:390]: This an be changed to `ChannelName: string` but then this field will be not lazy fetched anymore.
Instead the call to `channel, _ := slacker.client.GetConversationInfo(ev.Channel, true)` has to be performed
during the creation of the `MessageEvent`.

If this is okay then i will chaange it.

File: context.go (hunk 2 above)
Comment (k321rf) [reply context.go:65]: https://github.com/shomali11/slacker/pull/100#pullrequestreview-1019743388

File: context.go (hunk 1 above)
Comment (k321rf) [reply context.go:59]: https://github.com/shomali11/slacker/pull/100#pullrequestreview-1019743388

File: slacker.go (hunk 3 above)
Comment (arusso) [reply slacker.go:390]: You're right, and generally I would side on you with doing lazy fetching to save a few http calls. However in this case its not clear to me that the performance saved here is worth breaking consistency with the rest of the fields, and I can see someone using this getting tripped up, so I'm electing to keep things consistent.

If performance becomes an issue we can definitely revisit this.

**Review by k321rf — COMMENTED** [review]
Here is the eager implementation for fetching the channel name and username.

**Review by arusso — APPROVED** [review]

**Pull request merged**